cmd/argocd/commands/app.go

```diff
@@ -1617,3 +1619,85 @@ func NewApplicationPatchCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 	command.Flags().StringVar(&patch, "patch", "", "Patch")
 	return &command
 }
+
+func NewApplicationPatchResourceCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var patch string
+	var patchType string
+	var resourceName string
+	var namespace string
+	var kind string
+	var group string
+	var all bool
+	command := cobra.Command{
+		Use:   "patch-resource APPNAME",
+		Short: "Patch resource in an application",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 1 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			appName := args[0]
+
+			conn, appIf := argocdclient.NewClientOrDie(clientOpts).NewApplicationClientOrDie()
+			defer util.Close(conn)
+			ctx := context.Background()
+			resources, err := appIf.ManagedResources(ctx, &application.ResourcesQuery{ApplicationName: &appName})
+			errors.CheckError(err)
+			liveObjs, err := liveObjects(resources.Items)
+			errors.CheckError(err)
+			objectsToPatch := make([]*unstructured.Unstructured, 0)
+			for i := range liveObjs {
+				obj := liveObjs[i]
+				gvk := obj.GroupVersionKind()
+				if group != "" && kind != gvk.Group {
+					continue
+				}
+				if namespace != "" && namespace != obj.GetNamespace() {
+					continue
+				}
+				if resourceName != "" && resourceName != obj.GetName() {
+					continue
+				}
+				if kind == gvk.Kind {
+					copy := obj.DeepCopy()
+					objectsToPatch = append(objectsToPatch, copy)
+				}
+			}
+			if len(objectsToPatch) == 0 {
+				log.Fatal("No matching resource found to patch")
+			}
+			if len(objectsToPatch) > 1 && !all {
+				log.Fatal("Multiple resources match inputs. Use the --all flag to patch multiple resources")
+			}
+			for i := range objectsToPatch {
```

Comment:
Since `group` is optional, we need a check to verify that we are not patching two types of groups (e.g. knative.io/Service vs. corev1 Service). Can we loop through objects, to ensure group is consistent and error when it is not?